protocol/v2/ssv/runner/proposer.go

```diff
@@ -123,6 +133,19 @@ func (r *ProposerRunner) ProcessPreConsensus(ctx context.Context, logger *zap.Lo
 		zap.Uint64s("signers", getPreConsensusSigners(r.GetState(), root)),
 		fields.PreConsensusTime(r.measurements.PreConsensusTime()))
 
+	// Sleep the remaining proposerDelay since slot start, ensuring on-time proposals even if duty began late.
+	slotStartTime := r.BaseRunner.BeaconNetwork.EstimatedTimeAtSlot(duty.Slot)
+	timeIntoSlot := max(time.Since(time.Unix(slotStartTime, 0)), 0)
+	proposerDelayAdjusted := max(r.proposerDelay-timeIntoSlot, 0)
+	select {
+	case <-time.After(proposerDelayAdjusted):
+	case <-ctx.Done():
+		return ctx.Err()
+	}
```

Comment:
i think this could be more straightforward to follow:
```suggestion
	slotTime := r.BaseRunner.BeaconNetwork.EstimatedTimeAtSlot(duty.Slot)
	proposeTime := time.Unix(slotTime, 0).Add(r.proposerDelay)
	if timeLeft := time.Until(proposeTime); timeLeft > 0 {
		select {
		case <-time.After(timeLeft):
		case <-ctx.Done():
			return ctx.Err()
		}
	}
```